src/main/java/chess/service/ChessService.java

```diff
@@ -0,0 +1,117 @@
+        List<Square> squares = convertBoardToSquares(board);
+        squareDao.saveAll(squares, room.getId());
+        roomDao.update(room.getId(), webChessGame.getTurn());
+        return BoardDto.of(board, webChessGame.getTurn());
+    }
+
+    public BoardDto load(String roomName) {
+        Room room = roomDao.findByName(roomName)
+                .orElseThrow(() -> new IllegalStateException(NO_ROOM_MESSAGE));
+        ChessBoard chessBoard = loadChessBoard(room.getId());
+
+        return BoardDto.of(chessBoard.getPieces(), room.getTurn());
+    }
+
+    public BoardDto move(String roomName, MoveDto moveDto) {
+        Room room = roomDao.findByName(roomName)
+                .orElseThrow(() -> new IllegalStateException(NO_ROOM_MESSAGE));
+        WebChessGame webChessGame = WebChessGame.of(loadChessBoard(room.getId()), room.getTurn());
+        webChessGame.move(moveDto.getFrom(), moveDto.getTo());
+        roomDao.update(room.getId(), webChessGame.getTurn());
+        updateMovement(room.getId(), moveDto);
+
+        return BoardDto.of(webChessGame.getBoard(), webChessGame.getTurn());
+    }
+
+    private void updateMovement(long roomId, MoveDto moveDto) {
+        String fromPiece = squareDao.findByRoomIdAndPosition(roomId, moveDto.getFrom())
+                .orElseThrow(() -> new IllegalStateException(NO_SQUARE_MESSAGE))
+                .getPiece();
+        squareDao.update(roomId, moveDto.getFrom(), "empty");
+        squareDao.update(roomId, moveDto.getTo(), fromPiece);
+    }
+
+    private List<Square> convertBoardToSquares(Map<Position, Piece> board) {
+        return board.keySet().stream()
+                .map(position -> new Square(position.convertToString(), board.get(position).convertToString()))
+                .collect(Collectors.toList());
+    }
+
+    private ChessBoard loadChessBoard(long roomId) {
+        List<Square> squares = squareDao.findByRoomId(roomId);
+        if (squares.isEmpty()) {
+            throw new IllegalStateException(NO_SQUARES_MESSAGE);
+        }
+        Map<Position, Piece> board = new HashMap<>();
+        for (Square square : squares) {
+            Position position = Position.of(square.getPosition());
+            Piece piece = PieceFactory.convertToPiece(square.getPiece());
+            board.put(position, piece);
+        }
+        return new ChessBoard(() -> board);
+    }
+
+    public Status status(String name) {
+        Room room = roomDao.findByName(name)
+                .orElseThrow(() -> new IllegalStateException(NO_ROOM_MESSAGE));
+        ChessBoard chessBoard = loadChessBoard(room.getId());
+        WebChessGame webChessGame = WebChessGame.of(chessBoard, room.getTurn());
+
+        return webChessGame.status();
+    }
+
+    public void createRoom(String name) {
+        Room room = roomDao.findByName(name).orElse(null);
+        if (room == null) {
+            Room newRoom = new Room(name);
+            roomDao.save(newRoom);
+        }
+    }
```

Comment:
Optional을 반환하는데 null을 사용할 필요가 있을까요?
```suggestion

    public void createRoom(String name) {
 	Optional<Room> room = roomDao.findByName(name);
        if (room.isEmpty()) {
            Room newRoom = new Room(name);
            roomDao.save(newRoom);
        }
    }
```